lib/datadog/core/configuration/settings.rb

```diff
@@ -456,6 +456,7 @@ def initialize(*_)
 
           # NOTE: service also gets set as a side effect of tags. See the WORKAROUND note in #initialize for details.
           o.env Core::Environment::Ext::ENV_SERVICE
+          o.backup_env Core::Environment::Ext::ENV_OTEL_SERVICE
```

Comment:
If my previous suggestion is applied, this is what this line would look like:
```suggestion
          o.env [Core::Environment::Ext::ENV_SERVICE, Core::Environment::Ext::ENV_OTEL_SERVICE]
```